src/AspNetCore/Src/forwardinghandler.cxx

```diff
@@ -3141,45 +3145,54 @@ FORWARDING_HANDLER::TerminateRequest(
     bool    fClientInitiated
 )
 {
-    bool fAcquiredLock = FALSE;
-
-    if (TlsGetValue(g_dwTlsIndex) != this)
+    if (m_pApplication->QueryConfig()->QueryHostingModel() == HOSTING_IN_PROCESS) 
     {
-        DBG_ASSERT(TlsGetValue(g_dwTlsIndex) == NULL);
-        AcquireSRWLockShared(&m_RequestLock);
-        TlsSetValue(g_dwTlsIndex, this);
-        DBG_ASSERT(TlsGetValue(g_dwTlsIndex) == this);
-        fAcquiredLock = TRUE;
+        //
```

Comment:
I'll need to verify that this change works with the abort changes, however this seems like it would work.